### PR TITLE
.github: Add an issue template to record problems with the CI or debu…

### PR DESCRIPTION
**Pull request by philip-stoev**
…gging experience

### Motivation

  * This PR adds a feature that has not yet been specified.

The QA team would like to provide a venue for people to report general frustrations with the CI, the test frameworks or the test debugging experience in general.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/10-debugging-experience.yml
Comment (def-) [inline .github/ISSUE_TEMPLATE/10-debugging-experience.yml:10]:
```suggestion
name: "Internal: Feedback on CI, Test Frameworks and Debuggability"
```

File: .github/ISSUE_TEMPLATE/10-debugging-experience.yml
Comment (def-) [inline .github/ISSUE_TEMPLATE/10-debugging-experience.yml:20]:
```suggestion
        your frustration so that improvements can be made to make debugging easier in the future.
```

**Review by def- — CHANGES_REQUESTED** [review]
Looks good in general, just some minor nits

**Comment by philip-stoev** [comment]
@def- Apparently marking all the suggested changes as "resolved" is not sufficient to lift the "change requested" flag, so I have requested a re-review from you.

**Review by def- — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/10-debugging-experience.yml
Comment (benesch) [reply .github/ISSUE_TEMPLATE/10-debugging-experience.yml:10]: This doesn't look like it got addressed!

**Review by benesch — APPROVED** [review]
LGTM modulo pushing up fixes for @def-'s comments! Assume you have some unpushed fixes locally?

**Comment by philip-stoev** [comment]
Yes, sorry I had them locally all the time but just pushed them now.

File: .github/ISSUE_TEMPLATE/10-debugging-experience.yml
Comment (philip-stoev) [reply .github/ISSUE_TEMPLATE/10-debugging-experience.yml]: Yes sorry I forgot to force-push. It was all addressed.

**Pull request merged**
